e2e/specs/results-viewer.spec.ts

```diff
@@ -0,0 +1,168 @@
+  });
+
+  await test.step('And I click on the `Save and close` button', async () => {
+    await page.getByRole('button', { name: /save and close/i }).click();
+  });
+
+  await test.step('Then I should see a success toast notification', async () => {
+    await expect(page.getByText('The form has been submitted successfully.')).toBeVisible();
+  });
+
+  await test.step('When I go to the results viewer page', async () => {
+    await resultsViewerPage.goTo(patient.uuid);
+  });
+
+  await test.step('Then I should see the newly entered test results reflect in the results viewer', async () => {
+    await expect(page.getByRole('link', { name: /white blood cells/i })).toBeVisible();
+    await expect(page.getByText('12', { exact: true })).toBeVisible();
+    await expect(page.getByRole('link', { name: /red blood cells/i })).toBeVisible();
+    await expect(page.getByText('15', { exact: true })).toBeVisible();
+  });
+
+  await test.step('When I navigate to the `Visits` page', async () => {
+    visitsPage.goTo(patient.uuid);
+  });
+
+  await test.step('And I go to the `All encounters` tab', async () => {
+    await page.getByRole('tab', { name: /all encounters/i }).click();
+  });
+
+  await test.step('Then I should see the newly added lab test results included in the list', async () => {
+    await expect(
+      page.getByRole('cell', { name: /laboratory test results/i }).getByText('Laboratory Test Results'),
+    ).toBeVisible();
+  });
+
+  await test.step('When I launch the overflow menu of the created lab test result', async () => {
+    await page
+      .getByRole('button', { name: /options/i })
+      .nth(0)
+      .click();
+  });
+
+  await test.step('And I click on the `Edit` button', async () => {
+    await page.getByRole('menuitem', { name: /edit this encounter/i }).click();
+  });
+
+  await test.step('And I amend the values for "White blood cells" and "Red blood cells"', async () => {
+    await page.locator('#ManualInputWhiteBloodCellsid').clear();
+    await page.locator('#ManualInputWhiteBloodCellsid').fill('13');
+    await page.locator('#ManualEntryRedBloodCellsid').clear();
+    await page.locator('#ManualEntryRedBloodCellsid').fill('16');
+  });
+
+  await test.step('And I save the form', async () => {
+    await page.getByRole('button', { name: /save and close/i }).click();
+  });
+
+  await test.step('Then I should see a success toast notification', async () => {
+    await expect(page.getByText('The form has been submitted successfully.')).toBeVisible();
+  });
+
+  await test.step('When I revisit the `Results Viewer` page', async () => {
+    await resultsViewerPage.goTo(patient.uuid);
+  });
+
+  await test.step('Then I should see the updated results reflect in the results viewer', async () => {
+    await expect(page.getByRole('link', { name: /white blood cells/i })).toBeVisible();
+    await expect(page.getByText('13', { exact: true })).toBeVisible();
```

Comment:
This test is prone to fail when 13 is present elsewhere on the UI. I'm still not sure how we can assert these values within the results viewer but I think there should be a better way.
![ResultsViewer](https://github.com/openmrs/openmrs-esm-patient-chart/assets/38831673/f0483293-864d-4cbe-85b1-c5d9aa7f132b)

Reply:
@denniskigen @jayasanka-sack any thoughts?

Reply:
The best practice is to check the text that exists within the element instead of checking anywhere on the page. 

ex: The value inside of the 1st row's second column of the hematocrit table should be -> 42